maltoolbox/attackgraph/query.py

```diff
@@ -141,56 +153,23 @@ def get_attack_surface(
             attacker.id
         )
         for child in attack_step.children:
+            if skip_compromised and child.is_compromised_by(attacker):
+                continue
             if is_node_traversable_by_attacker(child, attacker) and \
                     child not in attack_surface:
-                attack_surface.append(child)
-    return attack_surface
-
-def update_attack_surface_add_nodes(
-        attacker: Attacker,
-        current_attack_surface: list[AttackGraphNode],
-        nodes: list[AttackGraphNode]
-    ) -> list[AttackGraphNode]:
-    """
-    Update the attack surface of an attacker with the new attack step nodes
-    provided to see if any of their children can be added.
-
-    Arguments:
-    attacker                - the Attacker whose attack surface is sought
-    current_attack_surface  - the current attack surface that we wish to
-                              expand
-    nodes                   - the newly compromised attack step nodes that we
-                              wish to see if any of their children should be
-                              added to the attack surface
-    """
-    logger.debug('Update the attack surface for Attacker "%s"(%d).',
-        attacker.name,
-        attacker.id)
-    attack_surface = current_attack_surface
-    for attack_step in nodes:
-        logger.debug(
-            'Determine attack surface stemming from "%s"(%d) '
-            'for Attacker "%s"(%d).',
-            attack_step.full_name,
-            attack_step.id,
-            attacker.name,
-            attacker.id
-        )
-        for child in attack_step.children:
-            is_traversable = is_node_traversable_by_attacker(child, attacker)
-            if is_traversable and child not in attack_surface:
                 logger.debug(
-                    'Add node "%s"(%d) to the attack surface of '
+                    'Add node "%s"(%d) to the attack surface for '
```

Comment:
I'm perfectly happy with my English, by the way.

Reply:
I didn't mean it as a syntax correction. I changed that to be on par with the docstring wording I had, "attack surface for", to indicate it is not THE attack surface of the attacker that you will get, it will depend on from_nodes whether it is. I.e. you will get AN attack surface for the attacker based on the arguments you supply in the function.